### PR TITLE
perf(benchmark): add IssueService and AuditorService service-layer be…

### LINKED ISSUE #1540: perf(benchmark): add IssueService and AuditorService service-layer benchmarks for zkatdlog driver
## What's missing

`transfer_test.go` has service-layer benchmarks for Transfer (serial + parallel). Issue and Auditor have nothing equivalent:

- `issue_test.go`  no benchmarks at all. `BenchmarkIssuer` in `issue/issuer_test.go` calls
  `issuer.GenerateZKIssue()` directly, skipping wallet, signer, audit-info encoding, and
  serialization.
- `auditor_test.go`  doesn't exist at the v1 package level.

## What needs to happen

1. `benchmark/setup.go`  add `IssueSetup` and `AuditCheckSetup` types that wire the existing  crypto materials in `SetupConfiguration` into ready-to-call services.
2. `mock/token_commitment_loader.go`  new counterfeiter mock for `TokenCommitmentLoader`   (used by `AuditorService`, defined at `auditor.go:26-28`, currently unmocked).
3. `issue_test.go`  add `BenchmarkIssueServiceIssue` + `TestParallelBenchmarkIssueServiceIssue`.
4. `auditor_test.go` new file with `BenchmarkAuditorServiceCheck` + `TestParallelBenchmarkAuditorServiceCheck`.

## Notes

- Issue benchmark is independent, no blockers.
- Auditor benchmark has a soft dependency on PR #1531  `AuditCheckSetup` needs  `GetTokenOutputs` to be restored for the end-to-end path to pass.
- `AuditCheckSetup` is also the first thing that exercises `AuditorService →
  TokenCommitmentLoader → audit.Auditor.Check` as a unit.

### Solution

Add `IssueSetup` and `AuditCheckSetup` helpers to `benchmark/setup.go`, a counterfeiter mock for `TokenCommitmentLoader`, and service-layer benchmarks for `IssueService.Issue()` and `AuditorService.AuditorCheck()` mirroring the existing Transfer benchmark pattern.

### PR DESCRIPTION
**Pull request by atharrva01**
closes #1540 

## What this does

The existing transfer benchmarks only cover one of the three token operations.
`IssueService.Issue()` and `AuditorService.AuditorCheck()` had no service-layer benchmarks  only a crypto-layer `BenchmarkIssuer` that calls `GenerateZKIssue()` directly and skips the full service path entirely.

This PR fills that gap.

## Changes

- **`benchmark/issue_setup.go`**  `IssueSetup` wires a real `IssueService` with mock wallet and signer, so each iteration exercises wallet lookup → signer resolution → ZK proof → audit-info encoding → serialization. `AuditCheckSetup` generates a real ZK issue action at setup time and feeds it into `AuditorService.AuditorCheck()` on each iteration.
 - **`mock/token_commitment_loader.go`**  new counterfeiter mock for `TokenCommitmentLoader`,  needed for `AuditorService` construction and forward-compatible with PR #1531.
- **`auditor.go`**  added `//go:generate` directive.
- **`issue_test.go`**  added serial + parallel benchmarks for `IssueService.Issue()`.
- **`auditor_test.go`**  new file with serial + parallel benchmarks for `AuditorService.AuditorCheck()`.

## Numbers (i7-12650H, WSL2)

```
BenchmarkIssueServiceIssue/32-BN254     136 ops   26.3 ms/op
BenchmarkIssueServiceIssue/64-BN254      80 ops   46.4 ms/op
BenchmarkIssueServiceIssue/32-BLS12      94 ops   40.6 ms/op

BenchmarkAuditorServiceCheck/32-BN254  19646 ops  185 µs/op
BenchmarkAuditorServiceCheck/32-BLS12   7060 ops  465 µs/op
```

All existing tests pass. `go vet` clean.

## Notes

- Issue benchmark is independent, no blockers.
- Auditor benchmark has a soft dependency on PR #1531, once `GetTokenOutputs` is restored,
  the mock is already wired to handle it.

### CODE REVIEW

**Comment by atharrva01** [comment]
hi @adecaro I added service-layer benchmarks for IssueService.Issue() and AuditorService.AuditorCheck(), covering full execution paths beyond the existing crypto-only benchmarks.

**Comment by adecaro** [comment]
@AkramBitar , please, review this one. Thanks 🙏

**Comment by adecaro** [comment]
Hi @atharrva01 , please, resolve the conflicts with the main branch 🙏

**Comment by adecaro** [comment]
overall, it looks good to me. Please, also update the documentation under `docs/drivers/benchmark/core/dlognogh` and the benchmarking scripts under `cmd/benchmarking`. 

Many many thanks 🙏

**Comment by atharrva01** [comment]
yess @adecaro , i was already working on that , the conflicts are now resolved

**Comment by atharrva01** [comment]
@adecaro , i've made both the changes you mentioned.  please have a look

File: docs/drivers/benchmark/core/dlognogh/dlognogh.md [hunk 1]
```diff
@@ -296,4 +298,72 @@ python run_benchmarks.py --benchName BenchmarkSender --timeout 4s --count 5
 - Running just one selected benchmark 5 times for no more than 4 seconds:
 ```shell
 python run_benchmarks.py --benchName BenchmarkSender --timeout 4s --count 5
-```
\ No newline at end of file
+```
+
+## Benchmark: `token/core/zkatdlog/nogh/v1#BenchmarkIssueServiceIssue`
```
Comment (adecaro) [inline docs/drivers/benchmark/core/dlognogh/dlognogh.md]: no need for this additional section unless there is something specific to it. The sections before should be enough to give the guidelines.

File: docs/drivers/benchmark/core/dlognogh/dlognogh.md [hunk 2]
```diff
@@ -38,6 +38,8 @@ Packages with benchmark tests:
 - `token/core/zkatdlog/nogh/v1/issue`: `BenchmarkIssuer` and `BenchmarkProofVerificationIssuer`
 - `token/core/zkatdlog/nogh/v1/validator`: `TestParallelBenchmarkValidatorTransfer`.
 - `token/core/zkatdlog/nogh/v1`: `BenchmarkTransferServiceTransfer` and `TestParallelBenchmarkTransferServiceTransfer`.
+- `token/core/zkatdlog/nogh/v1` (service layer): `BenchmarkIssueServiceIssue` and `TestParallelBenchmarkIssueServiceIssue` benchmark the full `IssueService.Issue()` path (wallet lookup, signer resolution, ZK proof generation, audit-info encoding, and serialization).
```
Comment (adecaro) [inline docs/drivers/benchmark/core/dlognogh/dlognogh.md]: please, remove `(service layer)` to be aligned with the line before this.

File: token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	issuerWallet := &drivermock.IssuerWallet{}
+	issuerWallet.GetSignerReturns(conf.IssuerSigner.Signer, nil)
+
+	walletService := &drivermock.WalletService{}
+	walletService.IssuerWalletReturns(issuerWallet, nil)
+
+	// GetAuditInfo is called twice inside Issue(): once per recipient and once
+	// for the issuer identity. Returning fixed bytes is sufficient because the
+	// audit info is stored in the metadata but not validated within Issue().
+	deserializer := &drivermock.Deserializer{}
```
Comment (adecaro) [inline token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go]: we need to instantiate the deserializer that is instantiated by the driver, otherwise the results are less precise of the behavior of the issuer service.

File: token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+	ppm, err := tokcommon.NewPublicParamsManagerFromParams(conf.PP)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed creating public params manager for audit check setup")
+	}
+
+	// MatchIdentity always returns nil so identity verification is a no-op.
+	// The benchmark focuses on the Pedersen commitment arithmetic in InspectOutput,
+	// not on identity-matching overhead.
+	deserializer := &drivermock.Deserializer{}
```
Comment (adecaro) [inline token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go]: same thing here.

File: token/core/zkatdlog/nogh/v1/auditor.go [hunk 5]
```diff
@@ -19,9 +19,15 @@ import (
 	"github.com/hyperledger-labs/fabric-token-sdk/token/core/zkatdlog/nogh/v1/validator"
 	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
 	"github.com/hyperledger-labs/fabric-token-sdk/token/services/logging"
+	token2 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
 	"go.opentelemetry.io/otel/trace"
 )
 
+//go:generate counterfeiter -o mock/token_commitment_loader.go -fake-name TokenCommitmentLoader . TokenCommitmentLoader
```
Comment (adecaro) [inline token/core/zkatdlog/nogh/v1/auditor.go]: we don't need this interface anymore per your previous code changes

**Comment by adecaro** [comment]
Hi @atharrva01 , left a few more comments on the PR. We are in the right direction. Thanks a lot for the effort 🙏

**Comment by atharrva01** [comment]
hi @adecaro , i have addressed all the reviews , please take a look

File: token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+	Request  *driver.TokenRequest
+	Metadata *driver.TokenRequestMetadata
+	Anchor   driver.TokenRequestAnchor
+}
+
+// NewAuditCheckSetup constructs an AuditCheckSetup from a SetupConfiguration.
+// It generates a real ZK issue action using the crypto layer directly, then
+// wires an AuditorService with the driver Deserializer so that identity
+// verification matches production behaviour.
+func NewAuditCheckSetup(conf *SetupConfiguration) (*AuditCheckSetup, error) {
```
Comment (adecaro) [inline token/core/zkatdlog/nogh/v1/benchmark/issue_setup.go]: either we put this in a different file auditor_setup.go or we change the name of the file issue_setup.go :)

**Comment by adecaro** [comment]
@atharrva01 , very last comment about the file name and I'm ready to approve and merge. Thanks much 🙏 😅

**Comment by atharrva01** [comment]
done @adecaro 🙏

**Review by adecaro — APPROVED** [review]
LGTM

**Comment by AkramBitar** [comment]
@atharrva01 thanks a lot for submitting this PR. 

Could you please check if we need to update this md file following adding your new benchmarks.
 https://github.com/hyperledger-labs/fabric-token-sdk/blob/main/docs/drivers/benchmark/core/dlognogh/dlognogh_architecture.md 
 This docs lists all the benchmarks that we have in the different layers. 

Thanks a lot.

**Comment by atharrva01** [comment]
hi @AkramBitar , i've updated dlognogh_architecture.md to add the missing benchmark/test entries in Layer 3, refreshed both the diagram and details, and removed the duplicate Transfer Validator Service section.
